tests/TestSearchPath.cc

```diff
@@ -42,15 +42,30 @@ TEST_CASE("Check SearchPath honors XDG_DATA_HOME", "[SearchPath]") {
     }
 }
 
+TEST_CASE("Check SearchPath defaults for unset XDG_DATA_DIRS", "[SearchPath]") {
+    setenv("XDG_DATA_HOME", "/does/not/exist", 1);
+    unsetenv("XDG_DATA_DIRS");
+
+    std::vector<std::string> result = get_search_path();
+
+    std::vector<std::string> expected;
+    if (is_directory("/usr/local/share/applications/"))
+        expected.push_back("/usr/local/share/applications/");
+    if (is_directory("/usr/share/applications/"))
+        expected.push_back("/usr/share/applications/");
+
+    REQUIRE(result == expected);
```

Comment:
This test heavily relies on and changes the environment. This has to be done with great care. The `expected` variable has some edge cases, but you've handled them. The overridden `$XDG_DATA_HOME` and `$XDG_DATA_DIRS` will leak into other tests. The current tests also do this (they come from [the r2.18 version](https://github.com/enkore/j4-dmenu-desktop/blob/r2.18/src/TestSearchPath.cc#L12)), so I'll tolerate it here, but I should maybe change these tests later. But "if it ain't broke, don't fix it."

Reply:
Actually, aside from the awkwardness, I realized that due to depending on the environment, this test does _most often_ not prove what it should. Mainstream distros (Arch, Ubuntu) do not create `/usr/local/share/applications/` by default (you have to do it yourself), so on a vanilla install, this test will pass even if you revert the changes in src/SearchPath.cc.